colaraz/lms/static/js/discussion/colaraz_discussion_filters.js

```diff
@@ -304,4 +305,63 @@ function reLoadDropDowns() {
 
 $(document).on('click', '.forum-new-post-form .submit', function () {
     reLoadDropDowns();
-});
\ No newline at end of file
+});
+
+// search all topics code down here
+// on click of search button
+$(document).on('click', '#cz-search-posts', function (e) {
+    nextUrlSearchPosts = colarazSearchPosts + '&text_search=' + $('#all-discussions-filter').val();
+    $.ajax({
+        type: 'GET',
+        url: nextUrlSearchPosts,
+        success: function (resp) {
+            var parentId = '#all-discussions-main';
+            var loadMoreClassName = 'all-discussions-loadmore';
+            var allDiscussionDropDown = '';
+            nextUrlSearchPosts = resp['pagination']['next'];
+
+            resp['results'].forEach(element => {
+                allDiscussionDropDown += createDiscussionElement(element);
+            });
+
+            if (allDiscussionDropDown) {
+                delNoPostsElement();
+                $('li .' + loadMoreClassName).remove();
+                $(parentId).html(allDiscussionDropDown);
+            } else {
+                 $(parentId).html(NoPostsElementAdd());
+            }
+
+
+            if (resp['pagination']['next'] != null) {
+                $(parentId).append(loadMoreButtonElement(loadMoreClassName));
+            }
+        },
+        error: function (resp) {
+            console.error(`All Discussions in Discussion API gives error: ${resp.Message}`);
+        },
+    });
+
+    e.stopPropagation();
+    e.preventDefault();
+
+});
+
+//reload all dicussions again
+$(document).on('click', '#cz-search-cancel', function (e) {
+    $('#all-discussions-filter').val('');
+    nextUrlAllDiscussions = colarazDiscussionAllThreadsUrl;
```

Comment:
Why are we re-assigning the value here?

Reply:
when user press load more it gets updated according to pagination so we reset it to its previous state once cancel button is clicked.

Reply:
Okay.